### DISCUSSION TITLE
Is there a better solution or plugin for converting px in `w-[50px]` to rem or vw?

### DISCUSSION THREAD

**Discussion by wangrongding** [tailwindlabs/tailwindcss#9869] [Ideas]
Is there a better solution or plugin for converting px in `w-[50px]` to rem or vw?  

<img width="406" alt="image" src="https://user-images.githubusercontent.com/42437658/202637337-89f22b2c-cbf4-4967-a385-ac1b493a3115.png">

I'm using vue3 with the `postcss-px-to-viewport` library, but it can't handle `w-[50px]` between lines, only properties with px in the style tag🥲

**Comment by wangrongding** [comment]
I am waiting for a good solution here~🤝

It is not a good method to write like this, because there are many properties. ↓
<img width="1133" alt="image" src="https://user-images.githubusercontent.com/42437658/202654570-7e0d3373-922a-49c3-82d1-900421671dbd.png">

**Reply by benoitdelorme** [reply]
I'm doing this:
```
const plugin = require('tailwindcss/plugin')
const postcss = require("postcss")
const designWidth = 1440

[...]

plugin.withOptions(() => {
    return function ({ addVariant }) {
        addVariant('vw', ({container}) => {
            let styles = []
            
            container.walkRules((rule) => {
                rule.walkDecls((decl) => {
                    styles.push({
                        value: decl.value.split('px')[0],
                        prop: decl.prop
                    })
                })
                rule.removeAll()
                
                styles.forEach(style => {
                    rule.prepend(postcss.decl({prop: style.prop, value: `${style.value * 100 / designWidth}vw`}))
                });
            });
        })
    };
})
```

You can use it like `vw:w-[300px]`, `vw:mt-[100px]` etc...

**Comment by wongjn** [comment]
Are you running `postcss-px-to-viewport` in the same pipeline that Tailwind is running in? Try including the plugin after Tailwind.

**Reply by wangrongding** [reply]
@wongjn I am now officially using Nuxt3. I don't think the method you mentioned provides a configuration solution, right?🥲🥲🥲

**Reply by wongjn** [reply]
If you have a similar setup for Tailwind as the [Tailwind Nuxt 3 integration guide](https://tailwindcss.com/docs/guides/nuxtjs), you should be able to add `postcss-px-to-viewport` after the Tailwind CSS PostCSS plugin.

**Reply by wangrongding** [reply]
🌸Thanks for your answer, if you don't consider whether it's `Nuxt` or not, may I ask how to set `postcss-px-to-viewport` to tailwind after it?

**Reply by wongjn** [reply]
If one runs Tailwind via PostCSS, one should be able to add `postcss-px-to-viewport` as a plugin after Tailwind,  via configuring `postcss.config.js` or similar.

**Reply by herman7** [reply]
This works well with tailwind version v3, I didn't find a good way to config it well in v4 after a long time to try

**Comment by ursazoo** [comment]
hi guys, any good solutions?

**Reply by wongjn** [reply]
Try https://github.com/tailwindlabs/tailwindcss/discussions/9869#discussioncomment-4174990
